### DISCUSSION TITLE
Make temp_allocator assert if it was not free_all'd at program termination

### DISCUSSION THREAD

**Discussion by xzores** [odin-lang/Odin#2754] [Proposals]
Make temp_allocator assert if it was not free_all'd at program termination, because of the change from ring buffer to a growing arena it might reduce memleak errors from new and old odin users.

**Comment by Kelimion** [comment]
`context.temp_allocator` can be reassigned, so there's no guarantee that the one assigned at the end of the program has been in use throughout.

**Comment by xzores** [comment]
No, but at that point you have actively changed it, that means you properly know that you would need to free it.

**Comment by Kelimion** [comment]
Unless you know you don't need to because the OS will free any allocations when the program exits, and now you're suddenly confronted with an assert to tell you of a purposeful leak.

By all means add leak checking to your own `main()` and dispatch to a `_main()` that's the actual body of your program, but please remember that leak checking is something not everyone does - sometimes on purpose, and sometimes depending on whether it's a release or development build of the software.

Checking whether `temp_allocator` has any allocations left on it and asserting would annoy a good few people while the people who benefit from it can do so themselves in their `main()`.
